op-node/p2p/pings_test.go

```diff
@@ -0,0 +1,79 @@
+package p2p
+
+import (
+	"context"
+	"errors"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"golang.org/x/exp/slog"
+
+	"github.com/libp2p/go-libp2p/core/peer"
+	"github.com/libp2p/go-libp2p/p2p/protocol/ping"
+
+	"github.com/ethereum-optimism/optimism/op-service/clock"
+	"github.com/ethereum-optimism/optimism/op-service/testlog"
+)
+
+func TestPingService(t *testing.T) {
+	peers := []peer.ID{"a", "b", "c"}
+	log, captLog := testlog.CaptureLogger(t, slog.LevelDebug)
+
+	testCase := 0
+	pingFn := PingFn(func(ctx context.Context, peerID peer.ID) <-chan ping.Result {
+		out := make(chan ping.Result, 1)
+		switch testCase % 3 {
+		case 0:
+			// success
+			out <- ping.Result{
+				RTT:   time.Millisecond * 10,
+				Error: nil,
+			}
+		case 1:
+			// fake timeout
```

Comment:
Would you want a `time.Sleep` in this fake timeout?

Reply:
We can fake the timeout more deterministically by simply not sending a response to the `out` channel.